### PR TITLE
Add `StagingTable` class

### PR DESCRIPTION
**Pull request by michael-the1**
Implementation details:
- Split the `Table` class into a `Table` and `DataVaultTable` class
- Create a new class `StagingTable` that inherits from `Table`
- Refactor all uses of `staging_table` and `staging_schema` with `StagingTable`

### CODE REVIEW

File: src/diepvries/table.py [hunk 1]
```diff
@@ -2,18 +2,78 @@
...
+)
 
 
 class Table(ABC):
+    """A generic table.
+
+    Abstract class Table. Holds common properties between all database tables.
+    """
+
+    # pylint: disable=too-few-public-methods
```
Comment (matthieucan) [inline src/diepvries/table.py:23]: I didn't know that one existed. We could also disable this check, WDYT?

File: src/diepvries/table.py [hunk 2]
```diff
@@ -2,18 +2,78 @@
...
+
+        Args:
+             schema: Schema name.
+             name: Table name.
+             extract_start_timestamp: Extract start timestamp.
+        """
+        physical_name = STAGING_PHYSICAL_NAME_SQL_TEMPLATE.format(
+            staging_table=name,
+            staging_table_suffix=extract_start_timestamp.strftime("%Y%m%d_%H%M%S"),
+        )
```
Comment (matthieucan) [inline src/diepvries/table.py:68]: While there, I'd be up for getting rid of `STAGING_PHYSICAL_NAME_SQL_TEMPLATE` and moving that string creation here. It's not used anywhere else

File: src/diepvries/table.py (hunk 1 above)
Comment (michael-the1) [reply src/diepvries/table.py:20]: This one is fine for me tbh, at least it forces you to make a conscious decision.

File: src/diepvries/table.py (hunk 2 above)
Comment (michael-the1) [reply src/diepvries/table.py]: Good idea 👍

File: src/diepvries/table.py [hunk 3]
```diff
@@ -41,31 +96,15 @@ def __init__(self, schema: str, name: str, fields: List[Field], *_args, **_kwarg
...
-        """Representation of a Table object as a string.
-
-        This helps the tracking of logging events per entity.
-
-        Returns:
-            String representation of a Table.
-        """
-        return f"{type(self).__name__}: {self.schema}.{self.name}"
+        # Set in DataVaultLoad
+        self.staging_table: StagingTable = None
```
Comment (matthieucan) [inline src/diepvries/table.py:107]: What about declaring `staging_table: StagingTable` at the class level, to avoid the potential type issues with `None`?

Comment (michael-the1) [reply src/diepvries/table.py]: Good idea, just applied it.

**Review by matthieucan — APPROVED** [review]

**Pull request merged**